improver_tests/utilities/test_GAMFit.py

```diff
@@ -0,0 +1,152 @@
+        {
+            "model_specification": {"term": ["l", [0], {}]},
+            "distribution": "gamma",
+            "link": "inverse",
+            "fit_intercept": False,
+        },  # check that inputs related to the GAM model design are initialised correctly
+    ],
+)
+def test__init__(kwargs):
+    """Test that the class initializes variables correctly."""
+    # Skip test if pyGAM not available.
+    pytest.importorskip("pygam")
+
+    # Define the default, then update with any differently specified inputs
+    expected = {
+        "model_specification": None,
+        "max_iter": 100,
+        "tol": 0.0001,
+        "distribution": "normal",
+        "link": "identity",
+        "fit_intercept": True,
+    }
+    expected.update(kwargs)
+    result = GAMFit(**kwargs)
+
+    for key in [key for key in kwargs.keys()]:
+        assert getattr(result, key) == kwargs[key]
+
+
+@pytest.mark.parametrize(
+    "test,model_specification",
+    [
+        (
+            "basic",
+            {
+                "term_1": ["l", [0], {}],
+                "term_2": ["s", [1], {}],
+                "term_3": ["te", [2, 3], {}],
+                "term_4": ["f", [4], {}],
+            },
+        ),  # Test that each type of GAM term can be created correctly
+        (
+            "with_kwargs",
+            {
+                "term_1": ["l", [0], {"lam": 0.8}],
+                "term_2": ["s", [1], {"n_splines": 10, "basis": "cp"}],
+            },
+        ),  # Test that kwargs are passed to the pyGAM terms correctly
+        (
+            "exception",
+            {
+                "term_1": ["l", [0], {}],
+                "term_2": ["kittens", [1], {}],
+            },
+        ),  # Test that an exception is raised when an unknown term is provided
+    ],
+)
+def test_create_pygam_model(test, model_specification):
+    """Test that this method correctly creates a pyGAM equation and raises an exception when
+    provided with a bad input."""
+    # Skip test if pyGAM not available.
+    pytest.importorskip("pygam")
+    from pygam import f, l, s, te
+
+    if test in ["basic", "with_kwargs"]:
+        if test == "basic":
+            expected = l(0) + s(1) + te(2, 3) + f(4)
+        elif test == "with_kwargs":
+            expected = l(0, lam=0.8) + s(1, n_splines=10, basis="cp")
+        result = GAMFit(model_specification).create_pygam_model()
+        assert result == expected
+    elif test == "exception":
+        expected_msg = (
+            "An unrecognised term has been included in the GAM model specification."
+        )
+        with pytest.raises(ValueError, match=expected_msg):
+            GAMFit(model_specification).create_pygam_model()
+
+
+def test_process():
+    """Test that the process method returns the expected results. Uses an example from the pyGAM quick start
+    documentation: https://pygam.readthedocs.io/en/latest/notebooks/quick_start.html#Fit-a-Model."""
```

Comment:
I think that it would be helpful to add a comment regarding the dataset e.g. "The "wage" dataset used consists of the features: Year, Age, Education (as a category) with the target being a fictitious value for the expected wage."